src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2171,7 +2171,7 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         foreach (var arm in switchExpression.SwitchArms)
                         {
                             var armConversion = this.Conversions.ClassifyImplicitConversionFromExpression(arm.Value, targetType, ref useSiteDiagnostics);
-                            if (!armConversion.IsImplicit)
+                            if (!armConversion.IsImplicit || !armConversion.IsValid)
```

Comment:
>IsImplicit [](start = 47, length = 10)

Would it make sense to adjust behavior of ```IsImplicit``` to return ```false``` when conversion doesn't exist?

Reply:
No.  From the language point of view, the conversion *does* exist, but produces an error.  Doing as you suggest would violate the language specification and it would be a (backward) breaking change in the language semantics.

Reply:
However, it is possible that the switch conversion itself should be considered to exist but not be valid when one of the element conversions exists but is not valid.  I'll make sure that is implemented and tested in a revision.

---
In reply to: [362603307](https://github.com/dotnet/roslyn/pull/40678#discussion_r362603307) [](ancestors = 362603307)